### ISSUE TITLE
Pocketsphinx Keyword Spotting plugins

### ISSUE THREAD

**Issue by aaronchantrill** [NaomiProject/Naomi#268]
## Detailed Description

Currently we are using PocketSphinx for keyword spotting, but that is only by building a special vocabulary including a keyword and some additional words to try and prevent the keyword listener from detecting everything as the keyword. There is, however, a special keyword mode in PocketSphinx that can be fine tuned to do a much better job of filtering keywords: https://cmusphinx.github.io/wiki/tutoriallm/#keyword-lists

Even though this task is fairly complex, I'm flagging this as a good first issue because it involves writing a pair of plugins, so it shouldn't affect Naomi's core.

## Context

I don't think the keyword list functionality existed in Pocketsphinx at the time that Jasper was first created. I think creating a special keyword mode pocketsphinx plugin could be extremely helpful in cutting down on the number of false positives.

## Possible Implementation

This would be a new keyword spotting plugin which would share the same acoustic model information with other pocketsphinx plugins, but would include the thresholds for all the keywords. There would also be a STTTrainer plugin which could be used with verified transcriptions to optimize the thresholds for maximum sensitivity with minimal false positives.

## Your Environment

* Version used:
* Environment name and version (e.g. PHP 5.4 on nginx 1.9.1):
* Server type and version:
* Operating System and version:
* Link to your project:

**Comment by aaronchantrill** [comment]
I'm working on updating the pocketsphinx plugins to use the new methods available through the bomboucher repository (which is the one we are using by default these days). Every time I create a new plugin related to pocketsphinx, though, I keep having to import .sphinxvocab, which imports .g2p, which imports .phonemeconversion, meaning that if we ever needed to update anything in any of those (switching to a different g2p converter than Phonetisaurus, for instance) we would have to make the changes in a bunch of different places (currently the pocketsphinx-stt, pocketsphinx_kws, and pocketsphinx_kws_train plugins). I'd like to move those three modules into the naomi directory, maybe under a new naomi.pocketsphinx package, so they will be available to any future plugin that need them.

Basically, I feel like pocketsphinx has proved its utility in this domain, and will continue to be used due to its speed and low resource use while providing good enough transcriptions for some purposes.

**Comment by aaronchantrill** [comment]
I created a plugin, "Pocketsphinx_KWS", which has this functionality. I was going to also write a STTTrainer plugin to help determine the best threshold for each keyword, but discovered that simply setting the threshold to max (80) always seems to work best.
